vendor-extra/JatsContentBundle/tests/ViewConverter/SubVisitorTest.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+namespace tests\Libero\JatsContentBundle\ViewConverter;
+
+use Libero\JatsContentBundle\ViewConverter\SubVisitor;
+use Libero\ViewsBundle\Views\View;
+use PHPUnit\Framework\TestCase;
+use tests\Libero\ContentPageBundle\ViewConvertingTestCase;
+use tests\Libero\ContentPageBundle\XmlTestCase;
+
+final class SubVisitorTest extends TestCase
+{
+    use ViewConvertingTestCase;
+    use XmlTestCase;
+
+    /**
+     * @test
+     * @dataProvider nodeProvider
+     */
+    public function it_does_nothing_if_it_is_not_a_jats_sub_element(string $xml) : void
+    {
+        $visitor = new SubVisitor($this->createFailingConverter());
+
+        $element = $this->loadElement($xml);
+
+        $newContext = [];
+        $view = $visitor->visit($element, new View(null), $newContext);
+
+        $this->assertNull($view->getTemplate());
+        $this->assertEmpty($view->getArguments());
+        $this->assertEmpty($newContext);
+    }
+
+    public function nodeProvider() : iterable
+    {
+        yield 'different namespace' => ['<sub xmlns="http://example.com">foo</sub>'];
+        yield 'different element' => ['<bold xmlns="http://jats.nlm.nih.gov">foo</bold>'];
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_nothing_if_is_not_the_sub_template() : void
+    {
+        $visitor = new SubVisitor($this->createFailingConverter());
+
+        $element = $this->loadElement('<sub xmlns="http://jats.nlm.nih.gov">foo</sub>');
+
+        $newContext = [];
+        $view = $visitor->visit($element, new View('template'), $newContext);
+
+        $this->assertSame('template', $view->getTemplate());
+        $this->assertEmpty($view->getArguments());
+        $this->assertEmpty($newContext);
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_nothing_if_there_is_already_text_set() : void
+    {
+        $visitor = new SubVisitor($this->createFailingConverter());
+
+        $element = $this->loadElement('<sub xmlns="http://jats.nlm.nih.gov">foo</sub>');
+
+        $newContext = [];
+        $view = $visitor->visit($element, new View(null, ['text' => 'bar']), $newContext);
+
+        $this->assertNull($view->getTemplate());
+        $this->assertSame(['text' => 'bar'], $view->getArguments());
+        $this->assertEmpty($newContext);
+    }
+
+    /**
+     * @test
+     */
+    public function it_sets_the_template_and_text_argument() : void
+    {
+        $visitor = new SubVisitor($this->createDumpingConverter());
+
+        $element = $this->loadElement(
+            <<<XML
+<jats:sub xmlns:jats="http://jats.nlm.nih.gov">
+    foo <jats:bold>bar</jats:bold> baz
```

Comment:
While we're not allowing mixed content in the spec, right to leave it working here (as we're allowing it to be changed).